### ISSUE TITLE
[Plato imports] I can remove a record from my suggestions

### ISSUE THREAD

**Issue by mietcls** [ugent-library/biblio-backoffice#1335]
# Enhancement
I can remove a record from my suggestions.

Part of https://github.com/ugent-library/biblio-backoffice/issues/1279 – needs to happen first.

## Success criteria

I can remove a record from my suggestions, when
- it is not mine
- it is already part of my bibliography
- an other reason

When I remove a suggestion...
- `[TBD]` My name is removed from the authors/supervisors/...
- `[TBD]` the record gets the status "rejected" and is visible to librarians
  - The rejected record can be completed by a librarian

This reason is added to the record, and can be checked by a librarian in https://github.com/ugent-library/biblio-backoffice/issues/1327

## Screenshot
<img width="564" alt="Screenshot 2023-12-04 at 09 47 44" src="https://github.com/ugent-library/biblio-backoffice/assets/1572885/d4838b50-d9fc-4124-a664-bd37bfe1b6d3">

## Implementation
```
<div id="modals" class="">
    <div id="modal-backdrop" class="d-block modal-backdrop fade show"></div>
    <div id="modal" class="d-block modal" tabindex="-1" role="dialog" aria-hidden="true">
        <div class="modal-dialog modal-dialog-lg modal-dialog-centered" role="document">
            <div class="modal-content">
                <div class="modal-header">
                    <h2 class="modal-title">Select reason and remove suggestion</h2>
                </div>
                <div class="modal-body">
                    <label class="c-label mb-5">Please select why you would like to remove the suggestion:</label>
                    <div class="form-check">
                        <input class="form-check-input" id="removeSuggestion1" type="radio" name="removeSuggestion" checked="checked">
                        <label class="form-check-label" for="removeSuggestion1">I am not an author, nor a supervisor for this publication.</label>
                    </div>
                    <div class="form-check">
                        <input class="form-check-input" id="removeSuggestion2" type="radio" name="removeSuggestion">
                        <label class="form-check-label" for="removeSuggestion2">I already have this publication in my bibliography.</label>
                    </div>
                    <div class="form-check">
                        <input class="form-check-input" id="removeSuggestion3" type="radio" name="removeSuggestion">
                        <label class="form-check-label" for="removeSuggestion3">Other reason</label>
                        <label class="c-label visually-hidden" for="otherReason">Other reason</label>
                        <input class="form-control" id="otherReason" type="text" />
                    </div>
                </div>
                <div class="modal-footer">
                    <div class="bc-toolbar">
                        <div class="bc-toolbar-left">
                            <button class="btn btn-link modal-close">Keep</button>
                        </div>
                        <div class="bc-toolbar-right">
                            <button class="btn btn-danger">Remove suggestion</button>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
```
https://biblio-backend-ugent-prototype-git-bootstr-307d84-ugent-library.vercel.app/styleguide/b-modals.html

# Out of scope
- Assigning rejected records to other people

**Comment by nicolasfranck** [comment]
For this purpose, I would suggest to:

* never clear the attribute `assigned_user_id` in the database. That way a librarian can always the original assigned user id
* use column `status` to differentiate:
  * `new`: newly created record, not evaluated by, and therefore visible by assigned user
  * `rejected`: "removed" by assigned user. This record will now move from this list to the list where librarian can see what records where rejected (see #1279).
  * `deleted`: "removed" by librarian (see #1327)

**Comment by mietcls** [comment]
Outdated
